core/src/main/kotlin/io/specmatic/core/pattern/JSONObjectPattern.kt

```diff
@@ -517,11 +527,7 @@ data class JSONObjectPattern(
         return if (anyPatternPaths.isNotEmpty()) {
             anyPatternPaths.map { anyPatternInfo ->
                 val formattedInfo = when {
-                    // Simple identifier (typeAlias) - needs braces
-                    anyPatternInfo.matches("^[a-zA-Z][a-zA-Z0-9]*$".toRegex()) -> "{$anyPatternInfo}"
-                    // Scalar type name - needs braces  
-                    anyPatternInfo in setOf("string", "number", "boolean") -> "{$anyPatternInfo}"
-                    // Complex path or already formatted - use as-is
+                    needsBraces(anyPatternInfo) -> "{$anyPatternInfo}"
```

Comment:
In fact, isn't this logic comparable to `private fun calculatePathForJSONObjectPattern(key: String, childValue: Value, objectPattern: JSONObjectPattern, resolver: Resolver): List<String> {`?